library/modules/Textures.cpp

```diff
@@ -41,205 +48,236 @@ static long g_window_borders_texpos_start = -1;
 
-long Textures::getThinBordersTexposStart() {
-    return g_thin_borders_texpos_start;
-}
+    DEFINE_VMETHOD_INTERPOSE(void, logic, ()) {
+        if (this->m_cur_step != this->cur_step) {
+            this->m_cur_step = this->cur_step;
+            if (this->m_cur_step == 2)
+                reset_texpos();
+        }
+        INTERPOSE_NEXT(logic)();
+    }
+
+  private:
+    inline static int m_cur_step = -2; // not valid state at the start
+};
+IMPLEMENT_VMETHOD_INTERPOSE(tracking_stage_adopt_region, logic);
+
+// reset point on Load Game
+struct tracking_stage_load_region : df::viewscreen_loadgamest {
+    typedef df::viewscreen_loadgamest interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(void, logic, ()) {
+        if (this->m_cur_step != this->cur_step) {
+            this->m_cur_step = this->cur_step;
+            if (this->m_cur_step == 1)
+                reset_texpos();
+        }
+        INTERPOSE_NEXT(logic)();
+    }
 
-long Textures::getMediumBordersTexposStart() {
-    return g_medium_borders_texpos_start;
+  private:
+    inline static int m_cur_step = -2; // not valid state at the start
+};
+IMPLEMENT_VMETHOD_INTERPOSE(tracking_stage_load_region, logic);
+
+// reset point on New Arena
+struct tracking_stage_new_arena : df::viewscreen_new_arenast {
+    typedef df::viewscreen_new_arenast interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(void, logic, ()) {
+        if (this->m_cur_step != this->cur_step) {
+            this->m_cur_step = this->cur_step;
+            if (this->m_cur_step == 0)
+                reset_texpos();
+        }
+        INTERPOSE_NEXT(logic)();
+    }
+
+  private:
+    inline static int m_cur_step = -2; // not valid state at the start
+};
+IMPLEMENT_VMETHOD_INTERPOSE(tracking_stage_new_arena, logic);
+
+static void install_reset_point() {
+    INTERPOSE_HOOK(tracking_stage_new_region, logic).apply();
+    INTERPOSE_HOOK(tracking_stage_adopt_region, logic).apply();
+    INTERPOSE_HOOK(tracking_stage_load_region, logic).apply();
+    INTERPOSE_HOOK(tracking_stage_new_arena, logic).apply();
 }
 
-long Textures::getBoldBordersTexposStart() {
-    return g_bold_borders_texpos_start;
+static void uninstall_reset_point() {
+    INTERPOSE_HOOK(tracking_stage_new_region, logic).remove();
+    INTERPOSE_HOOK(tracking_stage_adopt_region, logic).remove();
+    INTERPOSE_HOOK(tracking_stage_load_region, logic).remove();
+    INTERPOSE_HOOK(tracking_stage_new_arena, logic).remove();
 }
 
-long Textures::getPanelBordersTexposStart() {
-    return g_panel_borders_texpos_start;
+void Textures::init(color_ostream& out) {
+    install_reset_point();
+    DEBUG(textures, out).print("dynamic texture loading ready");
+
+    for (auto& [key, value] : g_static_assets) {
+        auto tile_w = TILE_WIDTH_PX;
+        auto tile_h = TILE_HEIGHT_PX;
+        if (key == "hack/data/art/pathable.png" || key == "hack/data/art/unsuspend.png") {
+            tile_w = 32;
+            tile_h = 32;
+        }
```

Comment:
this should be in the g_static_assets data structure so it can be maintained in one place